### ISSUE TITLE
Unable to remove offline node from HA cluster

### ISSUE THREAD

**Issue by kravciak** [rancher/rancher#32964]
In step 4. Delete the powered down node - the node is not deleted from rancher UI

Node status: `Cordoned - Waiting on node-controller`
Cleanup pod status (default namespace): `cattle-node-cleanup-t7bfg-pzltt`
```
0/6 nodes are available: 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had taint {node-role.kubernetes.io/etcd: true}, that the pod didn't tolerate.
```

The same test was executed last week and passed fine.

Rancher: single master-head (5d49b92)
Cluster: Custom cluster, 3 etcd, 1+1 control plane, 2 worker nodes

**Cluster information**
- Cluster type: Custom
- Machine type: AWS AMI: ami-012a79f775010f1b9, used add_custom_host job with default settings
- Kubernetes version: v1.21.1

**Comment by abhi1693** [comment]
@kravciak Is my issue the same as yours at https://github.com/rancher/rancher/issues/32913?

**Comment by kravciak** [comment]
Closing as duplicate

**Comment by derhally** [comment]
Not sure if it is the same, we just noticed several of our clusters ranging from EKS 1.16, 1.17, 1.18 showing nodes after the autoscaler scaled down.

![image](https://user-images.githubusercontent.com/13053772/122605803-f19f4580-d045-11eb-82b9-9f6cc7950af2.png)

**Comment by cbron** [comment]
Is this RKE2 only or both ?

**Comment by rmweir** [comment]
Followed the instructions mentioned in description. Originally it was not working due to a bug with etcd only nodes https://github.com/rancher/rancher/issues/32362.

Was unable to reproduce. I created a custom rke2 cluster with 1 controlplane, 3 wtcd, 2 workers, then added another controlplane and deleted the original. I did witness this issue https://github.com/rancher/rancher/issues/32891 but I was able to delete the node and it was removed leaving just one control plane:
![image](https://user-images.githubusercontent.com/19376037/128256870-53bb55f4-25e4-4193-aa2b-b47c4e02a651.png)

I did notice that it did not automatically delete machine object in dashboard UI associated with the same node. I was able to delete that machine manually as well.
